designer/server/src/main/scala/pl/touk/nussknacker/ui/services/NuDesignerApiAvailableToExpose.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.ui.services
+
+import org.reflections.Reflections
+import org.reflections.util.ConfigurationBuilder
+import pl.touk.nussknacker.ui.api.BaseEndpointDefinitions
+import pl.touk.nussknacker.ui.security.api.AuthCredentials
+import sttp.apispec.openapi.circe.yaml.RichOpenAPI
+import sttp.tapir.EndpointInput.Auth
+import sttp.tapir.docs.openapi.{OpenAPIDocsInterpreter, OpenAPIDocsOptions}
+import sttp.tapir.{Endpoint, auth}
+
+import scala.jdk.CollectionConverters._
+import java.lang.reflect.{Method, Modifier}
+import scala.util.Try
+
+object NuDesignerApiAvailableToExpose {
+
+  val name = "Nussknacker Designer API"
+
+  val openAPIDocsOptions: OpenAPIDocsOptions = OpenAPIDocsOptions.default
+    .copy(markOptionsAsNullable = true)
+
+  def generateOpenApiYaml: String = {
+    val endpoints = findApiEndpointsClasses().flatMap(findEndpointsInClass)
+    val docs = OpenAPIDocsInterpreter(openAPIDocsOptions).toOpenAPI(
+      es = endpoints,
+      title = NuDesignerApiAvailableToExpose.name,
+      version = ""
+    )
+
+    docs.toYaml
+  }
+
+  private def findApiEndpointsClasses() = {
+    val baseEndpointDefinitionsClass = classOf[BaseEndpointDefinitions]
+    val reflections = new Reflections(new ConfigurationBuilder().forPackages(baseEndpointDefinitionsClass.getPackageName))
+    reflections
+      .getSubTypesOf(baseEndpointDefinitionsClass).asScala
+      .toList
+      .sortBy(_.getName)
+  }
+
+  private def findEndpointsInClass(clazz: Class[_ <: BaseEndpointDefinitions]) = {
+    val endpointDefinitions = createInstanceOf(clazz)
+    clazz
+      .getDeclaredMethods.toList
+      .filter(isEndpointMethod)
```

Comment:
Maybe we should keep these values as an enum entries and use https://github.com/lloydmeta/enumeratum for the discovery purpose. WDYT?

Reply:
I don't know how to use `enumeratum` lib in this case. 
Yes, we want to get all endpoint definitions from a given class, but the class cannot converted to an object (we need to pass `Auth` obj) and each of the endpoints has a different returned type. All of these make `enumeratum` lib hard to be used in this particular case.